### PR TITLE
Split out lifecycle protocol

### PR DESCRIPTION
**Pull request by gaaclarke**
This puts iOS slightly more inline with Android without breaking current functionality.  It is more "clean" too since this new protocol is a nice logical grouping and it makes FlutterPlugin more grokable.

Related issue: https://github.com/flutter/flutter/issues/33071

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h [hunk 1]
```diff
@@ -18,107 +18,55 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol FlutterPluginRegistrar;
 @protocol FlutterPluginRegistry;
 
-/**
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h]: Man GitHub diffing is pretty crappy. But I'm assuming you're just copy pasting stuff around.

File: shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h [hunk 2]
```diff
@@ -18,107 +18,55 @@ NS_ASSUME_NONNULL_BEGIN
...
- * including headless `FlutterEngine` instances.
- *
- * This method is typically called from within an application's `AppDelegate` at
- * startup to allow for plugins which create additional `FlutterEngine` instances
- * to register the application's plugins.
- *
- * @param callback A callback for registering some set of plugins with a
- *     `FlutterPluginRegistry`.
+#pragma mark -
+/***************************************************************************************************
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h]: Do these long lines do anything special with doxygen or something?

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h]: Nope, they are ignored by doxygen.  It's for the benefit of someone scrolling through the file to see breaks between the top level objects.

File: shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h (hunk 1 above)
Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h]: Yep, just moved around.

**Comment by xster** [comment]
Still LGTM

**Pull request merged**